tests/test_reusable_executor.py

```diff
@@ -391,9 +391,9 @@ def test_numpy_dot_parent_and_child_no_freeze(self):
         pool is started in the parent.
         """
         a = np.random.randn(1000, 1000)
-        np.dot(a, a)
+        np.dot(a, a)  # trigger the thread pool init in the parent process
         executor = get_reusable_executor(max_workers=2)
-        executor.submit(np.dot, a, a)
+        executor.submit(np.dot, a, a).result()
```

Comment:
To make sure there is a correct error returned in case this test is broken, maybe we should put a timeout in `result` and use `shutdown(wait=True, kill_workers=True)` in case the timeout is reached?

```python
try:
    executor.submit(np.dot, a, a).result(timeout=60)
except TimeoutError:
    executor.shutdown(wait=True, kill_workers=True)
    raise
```